src/django_idom/hooks.py

```diff
@@ -30,3 +40,145 @@ def use_websocket() -> IdomWebsocket:
     if websocket is None:
         raise RuntimeError("No websocket. Are you running with a Django server?")
     return websocket
+
+
+_REFETCH_CALLBACKS: DefaultDict[FunctionType, set[Callable[[], None]]] = DefaultDict(
+    set
+)
+
+
+_Data = TypeVar("_Data")
```

Comment:
Functionally this seems redundant to me, is there a purpose behind this typevar?

Reply:
Yes, this is used to parametrize `Query`. This allows us to say that the return value of the `Query.data` will be the same type as is returned by the `query` function.